src/components/numeral-date/numeral-date.component.tsx

```diff
@@ -83,7 +186,23 @@ const NumeralDate = ({
 
   const refs = useRef(dateFormat.map(() => React.createRef()));
 
-  const [internalMessages, setInternalMessages] = useState({});
+  const [internalMessages, setInternalMessages] = useState<NumeralDateObject>({
+    ...((Object.fromEntries(
+      dateFormat.map((datePart) => [datePart, ""])
+    ) as Partial<FullDate>) as NumeralDateObject),
```

Comment:
I've tried to have a look to see if there is a way we could avoid using the type assertion for `Object.fromEntries`. However, given the [in-built ES2019 type definition for `Object.fromEntries`](https://github.com/microsoft/TypeScript/blob/c242d4a46b1085718d5607591b013288c9c88f9c/src/lib/es2019.object.d.ts) is the following:

```ts
fromEntries<T = any>(entries: Iterable<readonly [PropertyKey, T]>): { [k: string]: T };
```

This means the type of the object returned by `fromEntries` can only be an **index signature** `[x: string]: string` if the assertion is not used, and not of type:

```ts
type NumeralDateObject = 
| {
   dd: string;
   mm: string;
   yyyy: string;
} | {
   dd: string;
   mm: string;
} | {
   mm: string;
   yyyy: string;
}
```
as we want.

The only way round this would be to declare our own type definition for `Object.fromEntries` in `src/global.d.ts`, similar to [this example](https://dev.to/svehla/typescript-object-fromentries-389c). I would say the benefit to Carbon doing this would be better type safety wherever we use `Object.fromEntries` so we can catch further type bugs, but I'll leave this up to you if you think making this change would be beneficial or just add unnecessary complexity.

Reply:
That's a good point given the existing invariant we have 👍🏼